app/controllers/concerns/pagination.rb

```diff
@@ -3,9 +3,13 @@
 # Default options and configuration for pagination
 module Pagination
   extend ActiveSupport::Concern
+
+  ParamType = ActionController::Parameters
+
   included do
     def pagination_limit
-      params[:limit].present? ? params[:limit].to_i : 10
+      params[:limit].present? ? 
+      params.permit(limit: ParamType.integer & ParamType.gt(0))[:limit].to_i : 10
```

Comment:
You can move the `params.permit` to be a first statement. It would raise an exception before the method has time to return the value.

Reply:
This can be done on a single line now. The `.present?` check is redundant now because we no longer need the `to_i`, which causes problems with implicit conversions (i.e. `nil.to_i -> 0`, `'foo'.to_i -> 0`). See my original snippet suggestion (I believe it works as desired).